### DISCUSSION TITLE
Initial guidance on a PR to resolve the `Accept-Ranges` TODO

### DISCUSSION THREAD

**Discussion by itcarroll** [fsspec/filesystem_spec#1629] [Q&A] [answered]
I'd like to help resolve a `TODO` which could help `earthaccess` and the NASA Ocean Biology DAAC play nicely, at least if the resolution isn't beyond my skills (which do fall short of understanding async very well). Are there thoughts on a resolution here:

https://github.com/fsspec/filesystem_spec/blob/527728db774c9600821df76af6a78e53bca8d316/fsspec/implementations/http.py#L837-L840

The `TODO` has come to my attention because this DAAC does not accept ranges ... and is kinda aggressive 'bout it. If you try to GET a range, the response status is 416 (even if the range *is* within the file size). Everything goes downhill from there, including `HTTPFile.async_fetch_range` quietly returning `b""`. I don't have an informed opinion about the proper solution, but would try to help implement if there's any guidance. In the end, I'd like the following (which uses `HTTPFileSystem`) to automatically avoid range requests:

```
import earthaccess

earthaccess.login()
results = earthaccess.search_data(short_name="VIIRSJ1_L2_OC", count=1)
paths = earthaccess.open(results)
paths[0].read(8)
```

It looks like the plan is to use `HTTPStreamFile` rather than `HTTPFile` when `Accept-Ranges` is missing or `none`. Is there some subtlety to implementing that solution I should understand?

**Comment by itcarroll** [comment]
If a more targeted "fix" (for my use case) is advised, retrying with an HTTPStreamFile from here seems reasonable:

https://github.com/fsspec/filesystem_spec/blob/527728db774c9600821df76af6a78e53bca8d316/fsspec/implementations/http.py#L650

**Comment by martindurant** [comment]
I don't expect servers to necessarily tell the truth about whether they accept range requests or not :|

**Reply by itcarroll** [reply]
What about respecting "Accept-Ranges: 'none'" but not changing anything based on the lack of "Accept-Ranges"?

**Reply by martindurant** [reply, marked as answer]
That sounds reasonable. Trust the message when it is explicit.
